routing/index_road_graph.cpp

```diff
@@ -64,7 +64,12 @@ void IndexRoadGraph::GetEdges(Junction const & junction, bool isOutgoing, TEdgeV
   edges.clear();
 
   vector<SegmentEdge> segmentEdges;
-  m_starter.GetEdgesList(GetSegment(junction, isOutgoing), isOutgoing, segmentEdges);
+  for (Segment const & segment : GetSegments(junction, isOutgoing))
+  {
+    vector<SegmentEdge> tmpEdges;
```

Comment:
It'll save some number of allocations. Other approach is to stop clearing the vector in the GetEdgesList(). But it requires you to check all call places, as this will change the function contract.